cyclr.js

```diff
@@ -48,80 +53,115 @@ function buildFilters(datain) {
-              }
-              record.commitLineItem(fieldname);
+        }
+    }
+}
+
+function createRecord(datain) {
+    if (!datain.recordtype) {
+        var err = new Object();
+        err.status = "failed";
+        err.message = "missing recordtype";
+        return err;
+    }
+
+    var record = nlapiCreateRecord(datain.recordtype);
+    for (var fieldname in datain) {
+        if (datain.hasOwnProperty(fieldname)) {
+            if (fieldname != 'recordtype' && fieldname != 'id') {
+                var value = datain[fieldname];
+                if (value && typeof value == 'object') {
+                    if (value.length == undefined) {
+                        record.selectNewLineItem(fieldname);
+                        for (var sublistfield in value) {
+                            var sublistvalue = value[sublistfield];
+                            record.setCurrentLineItemValue(fieldname, sublistfield, sublistvalue);
+                        }
+                        record.commitLineItem(fieldname);
+                    } else {
+                        for (var i = 0; i < value.length; i++) {
+                            record.selectNewLineItem(fieldname);
+                            for (var sublistfield in value[i]) {
+                                var sublistvalue = value[i][sublistfield];
+                                record.setCurrentLineItemValue(fieldname, sublistfield, sublistvalue);
+                            }
+                            record.commitLineItem(fieldname);
+                        }
+                    }
+                } else {
+                    /**
+                    * Populate fields
+                    * sublists come in as objects that contain the line column values
+                    **/
+                    record.setFieldValue(fieldname, value);
+                }
             }
-          }
-        } else {
-          /**
-          * Populate fields
-          * sublists come in as objects that contain the line column values
-          **/
-          record.setFieldValue(fieldname, value);
         }
-      }
     }
-  }
 
-  var recordId = nlapiSubmitRecord(record);
-  return nlapiLoadRecord(datain.recordtype,recordId);
+    var recordId = nlapiSubmitRecord(record);
+    return nlapiLoadRecord(datain.recordtype, recordId);
 }
 
 function deleteRecord(datain) {
-  nlapiDeleteRecord(datain.recordtype, datain.id);
+    nlapiDeleteRecord(datain.recordtype, datain.id);
 }
 
 function updateRecord(datain) {
-  var record = nlapiLoadRecord(datain.recordtype, datain.id);
-  for (var fieldname in datain) {
-    if (datain.hasOwnProperty(fieldname)) {
-      if (fieldname != 'recordtype' && fieldname != 'id') {
-        var value = datain[fieldname];
-        if (value && typeof value != 'object') {
-          record.setFieldValue(fieldname, value);
+    var record = nlapiLoadRecord(datain.recordtype, datain.id);

```

Comment:
Is nlapiDateToString required when updating records to set the correct timezone?

Reply:
Yes. Upsert is not used by this client. It can be done later